lib/apiary/command/preview.rb

```diff
@@ -116,17 +90,49 @@ def write_generated_path(path, outfile)
         File.write(outfile, File.read(path))
       end
 
-      def generate_static(path)
-        File.open(preview_path(path), "w") do |file|
-          file.write(query_apiary(@options.api_host, path))
+      def generate
+        template = load_preview_template()
+
+        data = {
+          title: File.basename(@options.path, ".*"),
+          blueprint: load_blueprint()
+        }
+
+        template.result(binding)
+      end
+
+      def generate_static
+        preview_string = generate
+
+        File.open(preview_path, "w") do |file|
+          file.write preview_string
+          file.flush
           @options.output ? write_generated_path(file.path, @options.output) : open_generated_page(file.path)
         end
       end
 
+      def load_blueprint
+        file = File.open @options.path, "r"
+        file.read
+      end
+
+      def preview_path
+        basename = File.basename(@options.path, ".*")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.